services/elasticache/pkg/resource/replication_group/custom_update_api.go

```diff
@@ -87,7 +107,41 @@ func (rm *resourceManager) CustomModifyReplicationGroup(
 		return rm.updateShardConfiguration(ctx, desired, latest)
 	}
 
-	// no updates
+	return rm.modifyReplicationGroup(ctx, desired, latest)
+}
+
+// modifyReplicationGroup updates replication group
+// it handles properties that put replication group in
+// modifying state if these are supplied to modify API
+// irrespective of apply immediately.
+func (rm *resourceManager) modifyReplicationGroup(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (*resource, error) {
+	// Method currently handles SecurityGroupIDs, EngineVersion
+	// Avoid making unnecessary DescribeCacheCluster API call if both fields are nil in spec.
+	if desired.ko.Spec.SecurityGroupIDs == nil && desired.ko.Spec.EngineVersion == nil {
+		// no updates done
+		return nil, nil
+	}
+
+	// Get details using describe cache cluster to compute diff
+	latestCacheCluster, err := rm.describeCacheCluster(ctx, latest)
+	if err != nil {
+		return nil, err
+	}
+
+	// SecurityGroupIds
+	if rm.securityGroupIdsDiffer(desired, latest, latestCacheCluster) {
+		return rm.updateSecurityGroupIds(ctx, desired, latest)
+	}
+	// EngineVersion
+	if rm.engineVersionDiffer(desired, latest, latestCacheCluster) {
+		return rm.updateEngineVersion(ctx, desired, latest)
+	}
```

Comment:
Is there a reason you are making multiple API calls if both sec groups and engine version are different? Why not roll all updates into a single ModifyReplicationGroup API call?

Reply:
Updated the logic to make single modify API call.